Mappers/HttpMapper.php

```diff
@@ -191,36 +228,91 @@ protected function getHttpMapperConfig($urlName)
     }
 
     /**
-     * @param array $httpMapperConfig
-     * @return HttpMapper
+     * @param int $statusCode
+     * @throws HttpResourceNotFoundException
+     * @throws InfrastructureException
+     * @throws ValidationException
      */
-    private function setHttpMapperConfig($httpMapperConfig)
+    private function throwException(int $statusCode)
     {
-        $this->httpMapperConfig = $httpMapperConfig;
+        switch (true) {
+            case $statusCode == 404:
+                throw new HttpResourceNotFoundException('Http resource not found');
+
+                break;
+            case $statusCode >= 400 && $statusCode < 500:
+                throw new ValidationException([]);
+
+                break;
+            case $statusCode >= 500:
+                throw new InfrastructureException('InfrastructureException');
 
-        return $this;
+                break;
+        }
     }
 
     /**
-     * @param array $objectsParams
-     * @return PaginationCollection
+     * @return string
      */
-    protected function buildPaginationCollection(array $objectsParams)
+    protected function getUrl(): string
     {
-        $collection = new PaginationCollection();
-        $collection->setTotalResult(array_key_exists(
-            PaginationCollection::TOTAL_RESULTS, $objectsParams) ? $objectsParams[PaginationCollection::TOTAL_RESULTS] : 0
-        );
-        $collection->setLimit(array_key_exists(
-            PaginationCollection::LIMIT, $objectsParams) ? $objectsParams[PaginationCollection::LIMIT] : 0
-        );
-        $collection->setOffset(array_key_exists(
-            PaginationCollection::OFFSET, $objectsParams) ? $objectsParams[PaginationCollection::OFFSET] : 0
-        );
-        foreach ($objectsParams['items'] as $objectParams) {
-            $collection->push($this->buildObjectOptionalFields($objectParams));
-        }
+        return $this->getHttpMapperConfig(self::GET_URL);
+    }
+
+    /**
+     * @param array $identifiers
+     * @return string
+     */
+    protected function getOneUrl(array $identifiers)
```

Comment:
what this means? 
get one  url? 
can I get two url? or this is http get?

Reply:
it is clear from the context in new class